app/src/test/kotlin/compiler/e2e/NameResolutionErrorsTest.kt

```diff
@@ -1,5 +1,246 @@
+                        zm x: Liczba = 17
+                        wart x: Liczba = 18
+                    }
+                    
+                """,
+            Diagnostic.NameResolutionErrors.NameConflict::class
+        )
+        assertErrorOfType(
+            """
+                    czynność f() {
+                        zm x: Liczba = 17
+                        stała x: Liczba = 18
+                    }
+                    
+                """,
+            Diagnostic.NameResolutionErrors.NameConflict::class
+        )
+    }
+
+    @Ignore
+    @Test
+    fun `test conflicts (variables of different type)`() {
+        assertErrorOfType(
+            """
+                    czynność f() {
+                        zm x: Liczba = 17
+                        zm x: Czy = prawda
+                    }
+                    
+                """,
+            Diagnostic.NameResolutionErrors.NameConflict::class
+        )
+    }
+
+    @Ignore
+    @Test
+    fun `test conflicts (functions)`() {
+        assertErrorOfType(
+            """
+                    czynność f() {
+                        czynność g() -> Liczba {
+                            zwróć 17
+                        }
+                        czynność g() -> Liczba {
+                            zwróć 18
+                        }
+                    }
+                    
+                """,
+            Diagnostic.NameResolutionErrors.NameConflict::class
+        )
+    }
+
+    @Ignore
+    @Test
+    fun `test conflicts (functions with different return types)`() {
+        assertErrorOfType(
+            """
+                    czynność f() -> Liczba {
+                        czynność g() -> Czy {
+                            zwróć prawda
+                        }
+                        czynność g() -> Nic {
+                            zakończ
+                        }
+                    }
+                    
+                """,
+            Diagnostic.NameResolutionErrors.NameConflict::class
+        )
+    }
+
+    @Ignore
+    @Test
+    fun `test conflicts (functions with different signatures)`() {
+        assertErrorOfType(
+            """
+                    czynność f() -> Liczba {
+                        czynność g(a: Liczba) { }
+                        czynność g(a: Czy) { }
+                    }
+                    
+                """,
+            Diagnostic.NameResolutionErrors.NameConflict::class
+        )
+    }
+
+    // Assuming no functional features.
+
+    @Ignore
+    @Test
+    fun `test calling variables`() {
+        assertErrorOfType(
+            """
+                    czynność f() -> Czy {
+                        zm x: Czy = prawda
+                        zwróć x()
+                    }
+                    
+                """,
+            Diagnostic.NameResolutionErrors.VariableIsNotCallable::class
+        )
+    }
+
+    // I'm not 100% sure if the next two tests make sense
```

Comment:
In my opinion, they make sense - our language doesn't support using functions as variables at this point.